lms/services/blackboard_api.py

```diff
@@ -47,7 +73,15 @@ def get_token(self, authorization_code):
         )
 
     def request(self, method, path):
-        return self._http_service.request(method, self._api_url(path), oauth=True)
+        try:
+            return self._http_service.request(method, self._api_url(path), oauth=True)
+        except HTTPError as err:
+            error_dict = BlackboardErrorResponseSchema(err.response).parse()
+
+            if error_dict.get("message") == "Bearer token is invalid":
+                raise OAuth2TokenError() from err
+
+            raise
```

Comment:
When an access token is expired Blackboard sends `{"status": 401, "message": "Bearer token is invalid"}`. I'm not sure there's any way to detect this except by the human-readable error message.

`OAuth2TokenError` is already handled by a special exception view that sends a response that causes the frontend to show the **Authorize Hypothesis** dialog